packages/beacon-node/src/chain/opPools/singleAttestationPool.ts

```diff
@@ -122,6 +88,45 @@ export class SingleAttestationPool {
     return InsertOutcome.NewData;
   }
 
+  /**
+   * An aggregated attestations was seen, so we remove all SingleAttestations respective to that data
+   * as it's useless for us. In block production it'll prioritize aggregated attestations before reaching this pool.
+   */
+  seenAggregatedAttestation(
+    slot: Slot,
+    attDataRootHex: RootHex,
+    committeeIndex: CommitteeIndex,
+    aggregationBits: BitArray
+  ): void {
+    const committeeByIndexByRoot = this.committeeByIndexByRootBySlot.get(slot);
+    if (committeeByIndexByRoot == null) {
+      // no SingleAttestation for this slot
+      return;
+    }
+
+    const committeeByIndex = committeeByIndexByRoot.get(attDataRootHex);
+    if (committeeByIndex == null) {
+      // no SingleAttestation for this data root
+      return;
+    }
+
+    const committeeInfo = committeeByIndex.get(committeeIndex);
+    if (committeeInfo == null) {
+      // no SingleAttestation for this committee index
+      return;
+    }
+
+    // remove SingleAttestation for this committee index, we have it in AggregatedAttestationPool
+    const singleAttestations = committeeInfo.attestations;
+    // loop through singleAttestations instead of aggregationBits because after the 1st seen aggregated attestation,
+    // there is few/zero SingleAttestations left for the same committee index
```

Comment:
it does not take cpu time thanks for this

<img width="830" height="272" alt="Screenshot 2025-07-25 at 17 04 23" src="https://github.com/user-attachments/assets/82389e34-d8d2-470f-8614-378d2f5eee1a" />

forwarded messages and forwarded sent peers are the same for `beacon_aggregate_and_proof` topic
<img width="1701" height="296" alt="Screenshot 2025-07-25 at 17 05 41" src="https://github.com/user-attachments/assets/7019850f-deac-411f-b418-96a077c891c1" />

job wait time and job time are just the same

<img width="862" height="643" alt="Screenshot 2025-07-25 at 17 07 16" src="https://github.com/user-attachments/assets/f4f31864-1e96-4121-83c3-8ba0577acbbb" />